core/src/apps/cardano/seed.py

```diff
@@ -14,20 +14,62 @@
 
 
 class Keychain:
-    """Cardano keychain hard-coded to SEED_NAMESPACE."""
+    """Cardano keychain hard-coded to BYRON_SEED_NAMESPACE and SHELLEY_SEED_NAMESPACE."""
 
     def __init__(self, root: bip32.HDNode) -> None:
         self.root = root
+        self.byron_root = None
+        self.shelley_root = None
+
+        self.byron_namespace_length = len(BYRON_SEED_NAMESPACE)
```

Comment:
I don't really like the use of these lengths, but when Trezor guys redid the Keychain, they used this length instead of a constant with the value 2. I don't know if it makes any sense. Maybe only one constant `NAMESPACE_LENGTH` would be enough or not even that. Opinions?